pkg/fleet/installer/service/apm_sockets.go

```diff
@@ -0,0 +1,190 @@
+//
+// Note that we don't handle the case where sockets are configured in the agent configuration
+// but the old apm-inject is not setup. In this configuration, span drops are expected.
+func getSocketsPath(agentConfigPath, apmOldPath string) (string, string) {
+	apmSocket := apmInstallerSocket
+	statsdSocket := statsdInstallerSocket
+
+	// apmOldPath only exists if the agent is installed with the old apm-inject (deb/rpm)
+	// so we can return early if it doesn't
+	if _, err := os.Stat(apmOldPath); err == os.ErrNotExist {
+		return apmSocket, statsdSocket
+	}
+
+	rawCfg, err := os.ReadFile(agentConfigPath)
+	if err != nil {
+		log.Warn("Failed to read agent configuration file, using default installer sockets")
+		return apmSocket, statsdSocket
+	}
+
+	var cfg socketConfig
+	if err := yaml.Unmarshal(rawCfg, &cfg); err != nil {
+		log.Warn("Failed to unmarshal agent configuration, using default installer sockets")
+		return apmSocket, statsdSocket
+	}
+	if cfg.ApmSocketConfig.ReceiverSocket == apmOldSocket {
+		apmSocket = apmOldSocket
+	}
+	if cfg.DogstatsdSocket == statsdOldSocket {
+		statsdSocket = statsdOldSocket
+	}
+	return apmSocket, statsdSocket
+}
+
+// configureSocketsEnv configures the sockets for the agent & injector
+func configureSocketsEnv() error {
+	envFile := newFileMutator("/etc/environment", setSocketEnvs, verifySocketEnvs, verifySocketEnvs)
+	defer envFile.cleanup()
+	rollback, err := envFile.mutate()
+	if err != nil {
+		rollbackErr := rollback()
+		if rollbackErr != nil {
+			log.Warnf("Failed to rollback environment file: %v", rollbackErr)
+		}
+		return fmt.Errorf("error configuring sockets: %w", err)
+	}
+	return nil
+}
+
+// setSocketEnvs sets the socket environment variables
+func setSocketEnvs(envFile []byte) ([]byte, error) {
+	apmSocket, statsdSocket := getSocketsPath(agentConfigPath, apmInjectOldPath)
+	envs := map[string]string{
+		"DD_APM_RECEIVER_SOCKET": apmSocket,
+		"DD_DOGSTATSD_SOCKET":    statsdSocket,
+		"DD_USE_DOGSTATSD":       "true",
+	}
+	return addEnvsIfNotSet(envs, envFile)
+}
+
+// addEnvsIfNotSet adds environment variables to the environment file if they are not already set
+func addEnvsIfNotSet(envs map[string]string, envFile []byte) ([]byte, error) {
+	var buffer bytes.Buffer
+	buffer.Write(envFile)
+
+	if len(envFile) > 0 && envFile[len(envFile)-1] != '\n' {
+		buffer.WriteByte('\n')
+	}
+	for key, value := range envs {
+		// There's a slight gap where there exists an env var in the form of {prefix}{key} but not {key},
+		// but as our env vars all start with DD_ prefixing them won't cause any issue.
+		if !bytes.Contains(envFile, []byte(key+"=")) {
+			buffer.WriteString(fmt.Sprintf("%s=%s\n", key, value))
+		}
+	}
+	return buffer.Bytes(), nil
+}
+
+// verifySocketEnvs verifies that socket environment variables can be sourced
+func verifySocketEnvs(path string) error {
+	// Verify DD_USE_DOGSTATSD
+	cmd := exec.Command("/bin/sh", "-c", fmt.Sprintf(". %s && echo $DD_USE_DOGSTATSD", path))
```

Comment:
This is pretty shady, IDK if the reliability balance comes out positive. Maybe https://github.com/joho/godotenv is better but do we actually want this ? The current check also isn't perfect (env loaded by whatever the shell is vs systemd when its used).

Reply:
I can understand validate for docker since it can fail to reload but here we are just happening a few lines to a file and we can't actually test the real thing

Reply:
Removed in [78038f5](https://github.com/DataDog/datadog-agent/pull/25577/commits/78038f5faa5d7c6341e4fa2d68a826cbc1588c2a)